modules/elasticsearch-index/src/main/java/org/opencastproject/elasticsearch/index/ElasticsearchIndex.java

```diff
@@ -475,6 +555,85 @@ private void update(Series series) throws SearchIndexException {
     }
   }
 
+  /**
+   * Adds or updates the series in the search index. Uses a locking mechanism to avoid issues like Lost Update.
+   *
+   * @param ids
+   *          The ids of the series to add
+   * @param updateFunctionBundle
+   *          The functions that do the actual updating
+   * @param orgId
+   *          The organization the series belongs to
+   * @param user
+   *          The user
+   *
+   * @throws SearchIndexException
+   *          Thrown if unable to add or update the series.
+   */
+  public List<Optional<Series>> addOrUpdateSeriesBundle(List<String> ids,
+          List<Function<Optional<Series>, Optional<Series>>> updateFunctionBundle,
+          String orgId, User user) throws SearchIndexException {
+    List<Lock> lockBundle = new ArrayList<>();
+    for (String id: ids) {
+      lockBundle.add(this.locks.get(id));
+      lockBundle.get(lockBundle.size() - 1).lock();
+      logger.debug("Locked series '{}'", id);
+    }
+    try {
+      List<Optional<Series>> updatedSeriesOptBundle = new ArrayList<>();
+      List<Series> updatedSeriesBundle = new ArrayList<>();
+      boolean allPresent = true;
+      for (int i = 0; i < ids.size(); i++) {
+        Optional<Series> seriesOpt = getSeries(ids.get(i), orgId, user,
+                maxRetryAttemptsUpdate, retryWaitingPeriodUpdate);
```

Comment:
I agree with you and even if that was the original way how the index has been rebuilt I'll remove that whole step, working with updated elements for the rebuild instead of old elements and update functions.